### DISCUSSION TITLE
Is possible to set a default event parameter in Analytics?

### DISCUSSION THREAD

**Discussion by MFrat** [invertase/react-native-firebase#5241] [Q&A] [answered]
As [Firebase docs](https://firebase.google.com/docs/analytics/events?platform=android#set_default_event_parameters) describes, is possible to set some default event parameters, but a didn't find any mention of it in react-native-firebase docs, is it possible?

**Comment by mikehardy** [comment, marked as answer]
Interesting! I see this was added here:

- firebase-android-sdk BoM 25.9.0 https://firebase.google.com/support/release-notes/android#analytics_v17-5-0
- firebase-ios-sdk Pod 6.29.0 https://firebase.google.com/support/release-notes/ios#analytics_15

So the underlying SDKs added that feature in minimum versions we already require but we haven't had any PRs here to wrap that feature and expose it here

If you wanted to propose something I'd happily merge it in - believe it or not, development on this module isn't too hard, if you 

- fork/clone the repo and 
- follow https://github.com/invertase/react-native-firebase/blob/master/tests/README.md
- then you can go into the analytics e2e file here https://github.com/invertase/react-native-firebase/blob/master/packages/analytics/e2e/analytics.e2e.js and make one new test group as `describe.only(...` (note the `only`!) that just does a console.log in it's first `it(...)` test

When you run that with `tests:ios:test` or `tests:ios:test-reuse` (faster but does not reload native code) or the equivalent android one it will just run that one test each time, super fast.

I consider that "the workbench all set up with tools laid out" mode :-), at that point you have super focus on your area and you just code up one native bit, re-running it until it works via the test, then code up the other one, add it to the types in the index.d.ts file and it's ready

For analytics features even for a first time contributor might not take more than an hour or two? If you post any progress as a PR (even in draft form) I will happily work with you to get it to work

**Reply by MFrat** [reply]
Sure, thanks! I've opened a draft PR.
